core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1252,6 +1254,34 @@ private[spark] class DAGScheduler(
     execCores.map(cores => properties.setProperty(EXECUTOR_CORES_LOCAL_PROPERTY, cores))
   }
 
+  /**
+   * If push based shuffle is enabled, set the shuffle services to be used for the given
+   * shuffle map stage. The list of shuffle services is determined based on the list of
+   * active executors tracked by block manager master at the start of the stage.
+   */
+  private def prepareShuffleServicesForShuffleMapStage(stage: ShuffleMapStage) {
+    if (stage.shuffleDep.shuffleMergeEnabled) {
+      val mergerLocs = sc.schedulerBackend.getMergerLocations(
+        stage.shuffleDep.partitioner.numPartitions, stage.resourceProfileId)
+      if (mergerLocs.nonEmpty) {
+        stage.shuffleDep.setMergerLocs(mergerLocs)
+      } else {
+        stage.shuffleDep.setShuffleMergeEnabled(false)
+      }
+    } else {
+      // Disable Shuffle merge for the retry/reuse of the same shuffle dependency
+      stage.shuffleDep.setShuffleMergeEnabled(false)
+    }
+
+    if (stage.shuffleDep.shuffleMergeEnabled) {
```

Comment:
line 1276 to 1283 are just log statements for different conditions. Can't these be combined with 1263-1274?

Reply:
This is necessary because we are both enabling as well as disabling based on the conditions so we cannot merge these log statements into the above if/else conditions.

Reply:
Can't line 1277 go after 1267 and line 1280 go after 1269?